File: linealATramos.py
# Ian Mu;oz Nu;ez - Funcion lineal a tramos

import matplotlib.pyplot as plt
import numpy as np

n = 1000

v = np.linspace(-10, 10, n)

phi = np.zeros(n)
for i in range(0, n):
    if v[i] >= 1:
        phi[i] = 1
    elif v[i] > 0 and v[i] < 1:
        phi[i] = v[i]
    else:
        phi[i] = 0

plt.figure(1)
plt.grid()

plt.plot(v, phi, 'g', linewidth=2)

plt.title("Funcion lineal a tramos", fontsize=20)
plt.xlabel('v', fontsize=15)
plt.ylabel(r'$\phi$(v)', fontsize=15)

plt.show()

File: linealATramos.m
% Ian Mu;oz Nu;ez - Funcion lineal a tramos

close all
clear
clc

n = 1000;

v = linspace(-10, 10, n);

phi = [];
for i=1:n
    if v(i) >= 1
        phi(i) = 1;
    elseif v(i) > 0 && v(i) < 1
        phi(i) = v(i);
    else
        phi(i) = 0;
    end
end

figure(1)
hold on
grid on

plot(v, phi, 'g', 'LineWidth', 2)

title("Funcion lineal a tramos", 'FontSize', 20)
xlabel('v', 'FontSize', 15)
ylabel('\phi(v)', 'FontSize', 15)

